data/resources.go

```diff
@@ -2,95 +2,70 @@ package data
 
 import (
 	"context"
-	"time"
+	"embed"
 
-	"github.com/ONSdigital/log.go/v2/log"
+	"encoding/json"
 
 	"github.com/ONSdigital/dis-search-upstream-stub/models"
+	"github.com/ONSdigital/log.go/v2/log"
+	"github.com/pkg/errors"
 )
 
+//go:embed json_files/*.json
+var jsonFiles embed.FS
+
 // GetResources retrieves all the resources from the collection
 func (r *ResourceStore) GetResources(ctx context.Context, option Options) (*models.Resources, error) {
 	logData := log.Data{"option": option}
 	log.Info(ctx, "getting list of resources", logData)
 
-	// get resources count
-	numResources, err := r.getResourcesCount()
-	if err != nil {
-		log.Error(ctx, "failed to get resources count", err, logData)
-		return nil, err
-	}
-
-	// create and populate resourcesList
-	resourceList := make([]interface{}, numResources)
-	resourceList, err = r.populateResourceList(resourceList)
+	items, err := r.populateItems()
 	if err != nil {
+		logData["items"] = items
+		logData["count"] = len(items)
 		log.Error(ctx, "failed to populate resources list", err, logData)
 		return nil, err
 	}
 
 	resources := &models.Resources{
-		Count:      len(resourceList),
-		Items:      resourceList,
+		Count:      len(items),
+		Items:      items,
 		Limit:      option.Limit,
 		Offset:     option.Offset,
-		TotalCount: numResources,
+		TotalCount: len(items),
 	}
 
 	return resources, nil
 }
 
-// getResourcesCount returns the total number of resources stored
-func (r *ResourceStore) getResourcesCount() (int, error) {
-	return 2, nil
-}
-
-func (r *ResourceStore) populateResourceList(resourceList []interface{}) ([]interface{}, error) {
-	topics1 := []string{"a", "b", "c", "d"}
-	topics2 := []string{"a", "b", "e", "f"}
-	dateChanges := []string{"a change_notice", "a previous_date"}
-	tempResource1 := models.Standard{
-		URI:             "/a/temp/uri",
-		URIOld:          "/an/old/uri",
-		ContentType:     "api_dataset_landing_page",
-		CDID:            "ASELECTIONOFNUMBERSANDLETTERS123",
-		DatasetID:       "ASELECTIONOFNUMBERSANDLETTERS456",
-		Edition:         "a temporary edition",
-		MetaDescription: "a temporary description",
-		ReleaseDate:     time.Time{}.UTC(),
-		Summary:         "a temporary summary",
-		Title:           "a temporary title",
-		Topics:          topics1,
-		Language:        "string",
-		Survey:          "string",
-		CanonicalTopic:  "string",
+func (r *ResourceStore) populateItems() (items []interface{}, err error) {
+	logData := log.Data{}
```

Comment:
We're populating `logData` but not actually logging anything in this function. We can remove this now.

Reply:
👍